test/library/packages/Itertools/Itertools.chpl

```diff
@@ -192,4 +192,89 @@ module Itertools {
     }
   }
 
+
+
+  /*
+    Returns accumulated sums, differences, or results of other binary
+    operations (specified via the operation argument).
+
+
+    :arg arg: The iterable on which the accumulation is to be performed
+    :type arg: `array`
+
+    :arg operation: The operation which is to be performed for the
+    accumulation
+    :type operation: `operations (enum)`
+
+    :yields: Elements of the resultant array
+
+    :throws: ``IllegalArgumentError`` on non-array arguments
+
+
+    This iterator can be called in serial contexts.
+
+    .. note::
+      Be careful to pass ``real`` arrays if division is to be performed,
+      or the decimal part will be truncated.
+  */
+
+  enum operations { add, subtract, multiply, divide, bitwiseAnd, bitwiseOr, bitwiseXor }
+
+  iter accumulate(arg, operation) throws {
+
+    if !isArray(arg) then
+      throw new owned IllegalArgumentError(
+        "expected an array as argument");
+    else {
+      var result = arg[1];
+
+      for idx in arg.domain do
+        if idx == 1 then
+          yield result;
+        else {
+          select operation {
+
+            when operations.add do
+              result += arg[idx];
+
+            when operations.subtract do
+              result -= arg[idx];
+
+            when operations.multiply do
+              result *= arg[idx];
+
+            when operations.divide do
+              result /= arg[idx];
+
+            when operations.bitwiseAnd do
+              if result.type != int && result.type != bool then
+                throw new owned IllegalArgumentError(
+                  "bitwise operations supported only with boolean and integer types");
+              else
+                result &= arg[idx];
+
+            when operations.bitwiseOr do
+              if result.type != int && result.type != bool then
```

Comment:
This can be a compiler error since the types are known at compile-time.

Reply:
Yes, it could, but if I do something like [TIO](https://tio.run/##bY7BCoMwEETv@Yo5VhoE06OfIimEZFsFmw3RoiD59rTpRWl7W4Z9b8b2JtCYM/nnAxwomnlgP2GDcU5i5YjUChEiW9yYTybe5f5XYRPARCPZeU8/IbD05A/K@i2EYyxxmGn0xdQ1GmeUQ@mq/QuVAb/Q9QglkYQo27qmVhKqvmj51Vu1Ob8A), it still throws an error, as it checks every `select` case while compilation I guess.

Reply:
I am mistaken - that depends on the runtime value of `operations`.